### PR TITLE
Dimash branch

### PR DESCRIPTION
**Pull request by dimashisenov**
### Feature Preview

---

#### PR Checklist
Closes https://github.com/AppFlowy-IO/AppFlowy/issues/1820
- [x] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

**Comment by dimashisenov** [comment]
Currently resolving conflicts. Do not pay attention

File: frontend/appflowy_flutter/lib/plugins/document/document_page.dart [hunk 1]
```diff
@@ -22,6 +22,7 @@ import 'editor_styles.dart';
 import 'presentation/banner.dart';
 import 'presentation/plugins/grid/grid_view_menu_item.dart';
 import 'presentation/plugins/board/board_menu_item.dart';
+import '';
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/document_page.dart:25]: ?

File: frontend/appflowy_flutter/packages/appflowy_editor_plugins/lib/src/divider/dotted_divided_shortcut_event.dart [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:flutter/material.dart';
+import 'package:appflowy_editor_plugins/src/divider/dotted_divider_node_widget.dart';
+
+// insert divider into a document by typing three minuses.
+// ---
```
Comment (Xazin) [inline frontend/appflowy_flutter/packages/appflowy_editor_plugins/lib/src/divider/dotted_divided_shortcut_event.dart:6]: Wrong comment?

File: frontend/appflowy_flutter/packages/appflowy_editor_plugins/lib/src/divider/dotted_divided_shortcut_event.dart [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+ShortcutEvent insertDottedDividerEvent = ShortcutEvent(
+  key: 'DottedDivider',
+  command: "shift+underscore",
+  windowsCommand: "shift+underscore",
```
Comment (Xazin) [inline frontend/appflowy_flutter/packages/appflowy_editor_plugins/lib/src/divider/dotted_divided_shortcut_event.dart:11]:
```suggestion
  command: "_",
  windowsCommand: "_",
```

File: frontend/appflowy_flutter/packages/appflowy_editor_plugins/lib/src/divider/dotted_divided_shortcut_event.dart [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      .whereType<TextNode>();
+  if (textNodes.length != 1 || selection == null) {
+    return KeyEventResult.ignored;
+  }
+  final textNode = textNodes.first;
+  if (textNode.toPlainText() != '__') {
+    return KeyEventResult.ignored;
+  }
+  final transaction = editorState.transaction
+    ..deleteText(textNode, 0, 2) // remove the existing minuses.
```
Comment (Xazin) [inline frontend/appflowy_flutter/packages/appflowy_editor_plugins/lib/src/divider/dotted_divided_shortcut_event.dart:27]: Wrong comment

File: frontend/appflowy_flutter/packages/appflowy_editor_plugins/lib/src/divider/dotted_divider_node_widget.dart [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        return true;
+      };
+}
+
+class _DottedDividerWidget extends StatefulWidget {
+  const _DottedDividerWidget({
+    Key? key,
+    required this.node,
+    required this.editorState,
+  }) : super(key: key);
```
Comment (Xazin) [inline frontend/appflowy_flutter/packages/appflowy_editor_plugins/lib/src/divider/dotted_divider_node_widget.dart:28]:
```suggestion
    super.key,
    required this.node,
    required this.editorState,
  });
```

File: frontend/appflowy_flutter/packages/appflowy_editor_plugins/lib/src/divider/dotted_divider_node_widget.dart [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+class _DottedDividerWidgetState extends State<_DottedDividerWidget>
+    with SelectableMixin {
+  RenderBox get _renderBox => context.findRenderObject() as RenderBox;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.symmetric(vertical: 10),
+      child: Container(),
+    );
```
Comment (Xazin) [inline frontend/appflowy_flutter/packages/appflowy_editor_plugins/lib/src/divider/dotted_divider_node_widget.dart:46]: ?

File: frontend/appflowy_flutter/packages/appflowy_editor_plugins/lib/src/divider/dotted_divided_shortcut_event.dart (hunk 3 above)
Comment (dimashisenov) [reply frontend/appflowy_flutter/packages/appflowy_editor_plugins/lib/src/divider/dotted_divided_shortcut_event.dart:11]: This way of defining command did not work for me. Any suggestions?

Comment (Xazin) [reply frontend/appflowy_flutter/packages/appflowy_editor_plugins/lib/src/divider/dotted_divided_shortcut_event.dart:11]: You need to rebase your branch on `main`, or merge main into your branch with git pull.

Your branch is outdated and uses appflowy_editor v0.1.4, whereas main is on version v0.1.9.

**Comment by LucasXu0** [comment]
Hello, @dimashisenov. Are there any issues related to this PR? I'm not sure why we need to embed it.

**Pull request closed without merging**